src/Centreon/Application/Controller/AcknowledgementController.php

```diff
@@ -369,9 +329,16 @@ public function addHostAcknowledgement(
         if (!$contact->isAdmin() && !$contact->hasRole(Contact::ROLE_HOST_ACKNOWLEDGEMENT)) {
             return $this->view(null, Response::HTTP_UNAUTHORIZED);
         }
+        $content = json_decode((string) $request->getContent(), true);
+
+        // @todo check errors already sent for this
+        if ($content === false) {
+            throw new \InvalidArgumentException(_('Invalid data sent'));
+        }
```

Comment:
Regarding other controllers it should be an AcknowledgementException(_('Error when decoding sent data'))